### PR TITLE
Improve check of collinearity in vec2vec_rotmat

### PR DESCRIPTION
**Pull request by oesteban**
- [x] More strict check of collinearity of vectors.
  - [X] Fixed check for nans in the end.

### CODE REVIEW

File: dipy/core/geometry.py [hunk 1]
```diff
@@ -840,16 +840,17 @@ def vec2vec_rotmat(u, v):
     array([ 1.,  0.,  0.])
 
     """
-    norm_u_v = np.linalg.norm(u - v)
-    # return eye when u is the same with v
-    if np.linalg.norm(u - v) < np.finfo(float).eps:
+    w = np.cross(u, v)
```
Comment (stefanv) [inline dipy/core/geometry.py]: Documenting the logic behind the code would be helpful

**Comment by Garyfallidis** [comment]
@oesteban thank you for your PR as @stefanv suggests it would be nice to explain a bit the logic of your change. I would like to see a comment in the code or here explaining why your change is a more strict check of collinearity. After that I will merge this PR. Sounds good?

**Comment by oesteban** [comment]
I basically tried to replace workarounds by directly checking the cross product (which is at the core of finding the rotation matrix).

**Comment by arokem** [comment]
I am +1 for this change. The previous implementation would only work for unit vectors, but despite the documentation stating that the inputs should be unit vectors, there is nothing enforcing that they are. I think that this is more robust for distinguishing the antipodal vs. colinear case.

**Comment by arokem** [comment]
Since there were no further objections here, I am merging this.

**Pull request merged**